[Doc] Improve `useList`'s `filterCallback` prop description (#8404)
marmelab/react-admin: docs/useList.md

// ]
```

> ## `Return Value`

Reviewer: I'l -1 for this onr - return value isn't a prop name

    refetch, // a function that throws an error, as refetch doesn't make sense for local data
} = getGetList({ data });
```

## `filterCallback`

> Property for custom filter definition. Being able to apply local filters to the fetched data

Reviewer: ```suggestion
Property for custom filter definition. Lets you apply local filters to the fetched data.
```
